src/util/prolog/prologStringManipulation.ts

```diff
@@ -1,41 +1,122 @@
-const fetchFileExtension = /(.*)\.[^\.]*/;
-const fetchModuleDeclaration = /:-( |)module\([^)]*\) *\.[\r\t\f\v ]*\n?/g;
-const fetchPartialLineCommentsRegExp = /(^[^\n%]+)%[^\n\r]*(\r?\n)/gm;
-const fetchFullLineCommentsRegExp = /^[\r\t\f\v ]*%[^\n]*\n/gm;
-const fetchEmptyLinesRegExp = /\n(?:\s)*\n/gm;
-const fetchDoubleWhitespacesRegExp = /  */gm;
+import {fetchAllMatchesForAGroup} from "../regExpUtils";
+import builtInPredicates from "./builtInPredicates";
 
 export function removeFileExtension(path: string) {
+	const fetchFileExtension = /(.*)\.[^\.]*/;
+
 	const fetch: string[] | null = fetchFileExtension.exec(path);
 	if (fetch === null) { return ""; }
 	return fetch[1];
 }
 
 export function removeModuleDeclarations(prologProgram: string) {
+	const fetchModuleDeclaration = /:-( |)module\([^)]*\) *\.[\r\t\f\v ]*\n?/g;
+
 	return prologProgram.replace(fetchModuleDeclaration, "");
 }
 
 function removeFullLineComments(prologProgram: string) {
+	const fetchFullLineCommentsRegExp = /^[\r\t\f\v ]*%[^\n]*\n/gm;
+
 	return prologProgram.replace(fetchFullLineCommentsRegExp, "");
 }
 
 function removePartialLineComments(prologProgram: string) {
+	const fetchPartialLineCommentsRegExp = /(^[^\n%]+)%[^\n\r]*(\r?\n)/gm;
+
 	// group2 fetches the EOL standard in the file to fit both: Windows and UNIX
```

Comment:
I think it wouldn't hurt to type out the meaning of EOL. I, for my part, had to look it up. But it might just as well be common knowledge, which I didn't happen to have.

Reply:
I would have thought it is easier to find the term with EOL, but it isn't. I just tried googling it. I will include a link for more information on that since it isn't as easy to find as I thought it would be.